services/explorer/src/pages/Faucet/Faucet.effects.ts

```diff
@@ -0,0 +1,105 @@
+import { useSnackbar } from 'notistack';
+import { useCallback, useEffect, useState } from 'react';
+import { faucetUri } from '../../config';
+import { useExplorerNetwork } from '../../hooks/useExplorerNetwork';
+import { useExplorerLoader } from '../../hooks/useLoader';
+
+export enum RequestLimit {
+  ONE = 1,
+  TWO = 2,
+  FIVE = 5,
+  SEVEN = 7,
+}
+export const useFaucetEffects = () => {
+  const { currentNetwork } = useExplorerNetwork();
+  const { isLoading, initLoading, finishLoading } = useExplorerLoader();
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+  const [address, setAddress] = useState<string>('');
+  const [amount, setAmount] = useState<RequestLimit>(1);
+  const [displayToast, setDisplayToast] = useState<{
+    display: boolean;
+    variant?: 'success' | 'error' | 'warning' | undefined;
+    text?: string;
+  }>({ display: false, variant: undefined, text: '' });
+
+  const sendTokenRequest = useCallback(() => {
```

Comment:
It might be a good idea to move this outside, in a api folder. All API requests should be there.